### DISCUSSION TITLE
How to implement Agent with Semantic Kernel As of 08/08/2024?

### DISCUSSION THREAD

**Discussion by HuskyDanny** [microsoft/semantic-kernel#7972] [Q&A] [answered]
So now it seems we have various choices to implement agents. May I know the best possible choice for now that aligns the future path of the semantic kernel?

1. Autogen vs SK native agents? I used to read semantic kernel can integrate with Autogen but now I see SK's own implementation, can I assume SK team will support more on its own implementation?

2. Planner or Agent? Can I think of planner as ReAct based agent and Agent as function calling based agent in semantic kernel? And now it seems function calling based agent is more powerful?

**Comment by markat1** [comment]
+1  for @HuskyDanny  questions - let me add an extra question:

I have spend some time looking at some SK examples (agents etc) - what confuses me:

What is actually production ready  and what is not ?  because most of the examples contains either class or methods that has warnings on them saying they are only for test use or something like that.

**Reply by dmytrostruk** [reply, marked as answer]
> What is actually production ready and what is not ? because most of the examples contains either class or methods that has warnings on them saying they are only for test use or something like that.

@markat1 Just a small explanation about such warnings - when new feature is added to Semantic Kernel, we usually mark it as `Experimental`. It means that when we deliver this feature, our next step is to evaluate how it works, whether there are some issues reported with it etc. If we find some issues, we could add some changes to such feature that may break your compilation process as soon as you update to the new version of a library (e.g. we can change method signature or name). So, the purpose of such warnings is to warn users that this functionality can be updated/changed in the future in some way, but we usually don't change it drastically. So, it means that you can use it in your applications, but you should be ready for changes in newer versions if any.

**Reply by markat1** [reply]
I totally understand! thanks for the explanation - I guess I just wanted examples more divided into what's experimental and what's not, but I'm learning on the way! Notebooks are really great, and helps alot

**Reply by unaihuete93** [reply]
What about the difference between autogen and semantik kernel? Now that Semantic Kernel offers agent collaboration, which one should we use? or should we combine them? It is not really clear when to use what
